asteroid_data_processing.py

```diff
@@ -0,0 +1,417 @@
+import configparser
+import logging
+import os
+import re
+import time
+from datetime import datetime, timedelta
+
+import astropy.units as u
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+from astroplan import Observer
+from astropy import units as u
+from astropy.coordinates import AltAz, EarthLocation, SkyCoord
+from astropy.time import Time
+from astroquery.gaia import Gaia
+from flask import Flask, jsonify, request
+from matplotlib.patches import Circle
+
+app = Flask(__name__)
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+# logging.getLogger().setLevel(logging.WARNING)
+
+
+class bcolors:
+    HEADER = "\033[95m"
+    OKBLUE = "\033[94m"
+    OKCYAN = "\033[96m"
+    OKGREEN = "\033[92m"
+    WARNING = "\033[93m"
+    FAIL = "\033[91m"
+    ENDC = "\033[0m"
+    BOLD = "\033[1m"
+    UNDERLINE = "\033[4m"
+
+
+def timeit(func):
+    def timed(*args, **kwargs):
+        ts = time.time()
+        result = func(*args, **kwargs)
+        te = time.time()
+        logging.info(f"Function: {func.__name__} time: {round((te -ts)*1000,1)} ms)")
+        return result
+
+    return timed
+
+
+@timeit
+def read_config():
+    config_path = "config.ini"
+    try:
+        config = configparser.ConfigParser()
+        config.read(config_path)
+
+        RADIUS_FACTOR = config.getfloat("Parameters", "RADIUS_FACTOR")
+
+        MAX_STARS = config.getint("Parameters", "MAX_STARS")
+
+        MAX_DISTANCE = config.getfloat("Parameters", "MAX_DISTANCE")
+
+        QUERY_STARS_LIMIT = config.getfloat("Parameters", "QUERY_STARS_LIMIT")
+
+        MIN_DEG = config.getfloat("Parameters", "MIN_DEG")
+
+    except FileNotFoundError:
+        logging.exception("There is no config file named config.ini")
```

Comment:
Best practice is to use try/except only around the code which can actually raise a given exception.

```suggestion
    config = configparser.ConfigParser()
    try:
        config.read(config_path)
    except FileNotFoundError:
        logging.exception("There is no config file named config.ini")
    else:
        RADIUS_FACTOR = config.getfloat("Parameters", "RADIUS_FACTOR")

        MAX_STARS = config.getint("Parameters", "MAX_STARS")

        MAX_DISTANCE = config.getfloat("Parameters", "MAX_DISTANCE")

        QUERY_STARS_LIMIT = config.getfloat("Parameters", "QUERY_STARS_LIMIT")

        MIN_DEG = config.getfloat("Parameters", "MIN_DEG")
```